src/rad/core.clj

```diff
@@ -6,22 +6,44 @@
 
 ;; point is x & y position of the point (or cursor)
 (def point (atom [0 0]))
+(declare sync-frontend-cursor-to-point-atom! move-point-backwards! move-point-backwards)
+
+(defn move-point-backwards
+  "Returns point 'steps' steps backwards until it hits a beginning-of-line"
+  [buffer point steps]
+  (let [can-move-backwards? (fn [point-x]
+                              (> point-x 0))]
+    (loop [line (buffer (second point))
+           point-x (first point)
+           steps steps]
+      (if (and
+           (> steps 0)
+           (can-move-backwards? point-x))
+
+        (recur line (dec point-x) (dec steps))
+        [point-x (second point)]))))
+
+(defn move-point-backwards!
+  "Moves point 'steps' steps backwards, and syncs the UI cursor"
+  [steps] (do
+            (reset! point (move-point-backwards @rad.buffer/current-buffer @point steps))
+            (sync-frontend-cursor-to-point-atom!)))
 
-;; FIXME understand newlines
 (defn move-point-forward
   "Returns point 'steps' steps forward until it hits a newline."
-  [buffer point steps] (let [line (buffer (second point))
-                             point-x (first point)
-                             point-y (second point)
-                             can-move-forward? (fn [line p-x]
-                                                 (not (>= p-x (count line))))]
+  [buffer point steps]
+  (let [line (buffer (second point))
+        point-x (first point)
+        point-y (second point)
```

Comment:
Because this is used on line 35, it should be defined before. No biggie, but saves like 2 evaluation steps.